### ISSUE TITLE
Implement an image cache for the vault

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#252]

**Comment by mushon** [comment]
This should be optional: default - on

**Comment by dhowe** [comment]
@leoneckert this is a tricky one, but see what you can find out with a bit of research...

**Comment by leoneckert** [comment]
@dhowe I was actually just wondering if ADN stores the images (that's what a cache is if I understand correctly)?
I switched internet off/on when testing loading animation and was surprised all the ads show up...but I guess that's just my browsers cache..right?

**Comment by dhowe** [comment]
yes, adn does not store/cache any images at moment (this is the goal) ...
we need to be able to display stale ads even after the browser cache is cleared
i imagine there is some way to [disable the browser cache for testing](http://www.technipages.com/google-chrome-how-to-completely-disable-cache)

**Comment by leoneckert** [comment]
@dhowe do I understand the following right: If not otherwise configured, the cache is a default function of the browser to save bandwidth and make things quicker. When rendering the vault, chrome will, in any case, check it's own cache first for all the images and load them from there if it finds them.      
At least that seems to be the case already: when I empty my cache and then open my vault, Chrome will retrieve all the images from whichever servers they come from, display them in the vault and also store them in the cache - assumingly to load them from there the next time (also works offline).

I am guessing the cache gets updated when chrome notices a change in the original file (maybe by testing only the first few bits) from the actual server it came from. 

Oddly in the Chrome extension docs, only an option to [remove browsingData](https://developer.chrome.com/extensions/browsingData#method-removeCache) exists, not to read or create an extension specific one. 

Then there is the [chrome.storage](https://developer.chrome.com/extensions/storage#type-StorageArea) which we could use.

**Comment by dhowe** [comment]
That all sounds correct. I would start by adding an option (#639), don't worry about the UI for now, to download the files on export. I've started in core.js -- you can finish:

![screen shot 2016-11-22 at 3 02 46 pm](https://cloud.githubusercontent.com/assets/737638/20514043/cb8fe1ec-b0c4-11e6-9d08-c7ea1e611c30.png)

**Comment by ice-fly** [comment]
You might be able to [compress/chip](https://stackoverflow.com/questions/22329481/compressing-base64-data-uri-images) the advert into a [base64 Image URI](https://github.com/ChiefInnovator/image-to-data-uri-converter) at the time of collection and then fallback to that lowres whenever the advert dies.

**Comment by dhowe** [comment]
this would work for image ads, but we'd need some type of screen-cap for other types

**Comment by mneunomne** [comment]
This was implmented locally with the solution we used for the ZKM exhibition, but we never released this to the public. We could though?
